adminservice/src/main/java/nu/yona/server/goals/rest/ActivityCategoryController.java

```diff
@@ -40,18 +44,45 @@
 
 	@RequestMapping(value = "/{id}", method = RequestMethod.GET)
 	@ResponseBody
+	@JsonView(ActivityCategoryDTO.AdminView.class)
 	public HttpEntity<ActivityCategoryResource> getActivityCategory(@PathVariable UUID id)
 	{
 		return createOKResponse(activityCategoryService.getActivityCategory(id));
 	}
 
 	@RequestMapping(value = "/", method = RequestMethod.GET)
 	@ResponseBody
+	@JsonView(ActivityCategoryDTO.AdminView.class)
 	public HttpEntity<Resources<ActivityCategoryResource>> getAllActivityCategories()
 	{
 		return createOKResponse(activityCategoryService.getAllActivityCategories(), getAllActivityCategoriesLinkBuilder());
 	}
 
+	@RequestMapping(value = "/", method = RequestMethod.POST)
+	@ResponseBody
+	@JsonView(ActivityCategoryDTO.AdminView.class)
+	public HttpEntity<ActivityCategoryResource> addActivityCategory(@RequestBody ActivityCategoryDTO activityCategory)
+	{
+		activityCategory.setID(UUID.randomUUID());
```

Comment:
Given that it's a DTO, I'm not concerned about the public availability of the setter. Here, you _might_ move it to the service, but on the update, it has to be in the controller (at least, I feel it's a weird service contract to pass a DTO containing an ID and then the same ID in a parameter). Given that, I favor keeping this set also in the controller.